pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -101,29 +105,33 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 			Location: to.StringPtr(s.Scope.ClusterConfig.Location),
 			VirtualMachineProperties: &compute.VirtualMachineProperties{
 				HardwareProfile: &compute.HardwareProfile{
-					VMSize: compute.VirtualMachineSizeTypes("Standard_DS2_v2"),
+					VMSize: compute.VirtualMachineSizeTypes(vmSpec.Size),
 				},
 				StorageProfile: &compute.StorageProfile{
 					ImageReference: &compute.ImageReference{
-						Publisher: to.StringPtr("Canonical"),
-						Offer:     to.StringPtr("UbuntuServer"),
-						Sku:       to.StringPtr("18.04-LTS"),
-						Version:   to.StringPtr("latest"),
+						Publisher: to.StringPtr(vmSpec.Image.Publisher),
+						Offer:     to.StringPtr(vmSpec.Image.Offer),
+						Sku:       to.StringPtr(vmSpec.Image.SKU),
+						Version:   to.StringPtr(vmSpec.Image.Version),
 					},
 					OsDisk: &compute.OSDisk{
+						OsType:       compute.OperatingSystemTypes(vmSpec.OSDisk.OSType),
 						CreateOption: compute.DiskCreateOptionTypesFromImage,
-						DiskSizeGB:   to.Int32Ptr(64),
+						DiskSizeGB:   to.Int32Ptr(vmSpec.OSDisk.DiskSizeGB),
+						ManagedDisk: &compute.ManagedDiskParameters{
+							StorageAccountType: compute.StorageAccountTypes(vmSpec.OSDisk.ManagedDisk.StorageAccountType),
+						},
 					},
 				},
 				OsProfile: &compute.OSProfile{
 					ComputerName:  to.StringPtr(vmSpec.Name),
-					AdminUsername: to.StringPtr("azureuser"),
+					AdminUsername: to.StringPtr("capi"),
```

Comment:
Can we move this into defaults instead and then reference that `const` here and in the public key path below?

Reply:
Done. Thanks for the review